src/components/Table/GridTable.tsx

```diff
@@ -1306,119 +1279,13 @@ function maybeDarken(color: string | undefined, defaultColor: string): string {
   return color ? tinycolor(color).darken(4).toString() : defaultColor;
 }
 
-// Get the rows that are already in the toggled state, so we can keep them toggled
-function getCollapsedRows(persistCollapse: string | undefined): string[] {
-  if (!persistCollapse) return [];
-  const collapsedGridRowIds = localStorage.getItem(persistCollapse);
-  return collapsedGridRowIds ? JSON.parse(collapsedGridRowIds) : [];
-}
-
-/**
- * A custom hook to manage a list of ids.
- *
- * What's special about this hook is that we manage a stable identity
- * for the `toggleId` function, so that rows that have _not_ toggled
- * themselves on/off will have an unchanged callback and so not be
- * re-rendered.
- *
- * That said, when they do trigger a `toggleId`, the stable/"stale" callback
- * function should see/update the latest list of values, which is not possible with a
- * traditional `useState` hook because it captures the original/stale list identity.
- */
-function useToggleIds(
-  rows: GridDataRow<Kinded>[],
-  persistCollapse: string | undefined,
-): readonly [string[], GridCollapseContextProps, GridCollapseContextProps] {
-  // Make a list that we will only mutate, so that our callbacks have a stable identity.
-  const [collapsedIds] = useState<string[]>(getCollapsedRows(persistCollapse));
-  // Use this to trigger the component to re-render even though we're not calling `setList`
-  const [tick, setTick] = useState<string>("");
-
-  // Checking whether something is collapsed does not depend on anything
-  const isCollapsed = useCallback(
-    (id: string) => collapsedIds.includes(id),
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [],
-  );
-
-  const collapseAllContext = useMemo(
-    () => {
-      // Create the stable `toggleCollapsed`, i.e. we are purposefully passing an (almost) empty dep list
-      // Since only toggling all rows required knowledge of what the rows are
-      const toggleAll = (_id: string) => {
-        // We have different behavior when going from expand/collapse all.
-        const isAllCollapsed = collapsedIds[0] === "header";
-        collapsedIds.splice(0, collapsedIds.length);
```

Comment:
The core of this "what does collapse all mean for a header" logic moved into `RowState`.